src/core/ext/filters/client_channel/retry_filter_legacy_call_data.cc

```diff
@@ -2020,8 +2020,10 @@ void RetryFilter::LegacyCallData::OnRetryTimer() {
 void RetryFilter::LegacyCallData::OnRetryTimerLocked(
     void* arg, grpc_error_handle /*error*/) {
   auto* calld = static_cast<RetryFilter::LegacyCallData*>(arg);
-  calld->retry_timer_handle_.reset();
-  calld->CreateCallAttempt(/*is_transparent_retry=*/false);
+  if (calld->retry_timer_handle_.has_value()) {
```

Comment:
Would you mind leaving a small comment in the code about why is this condition needed?

Reply:
I don't think a comment is actually necessary.  We use this pattern basically everywhere we do callbacks, because of exactly this race condition.  It was just a bug that it wasn't done here.